src/components/Nav/SiteMap.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+
+const DIVIDE_CATEGORIES = 6;
+
+function SiteMap(props) {
+  useEffect(() => {
+    fetch('http://localhost:3000/data/navCategories.json')
+      .then(response => response.json())
+      .then(result => {
+        const addedEmptyCategories = 2 * DIVIDE_CATEGORIES - result.length;
+        if (
+          result.length > DIVIDE_CATEGORIES &&
+          result.length < 2 * DIVIDE_CATEGORIES
+        ) {
+          for (let i = 0; i < addedEmptyCategories; i++) {
+            result.push({});
+          }
+        }
+        setCategories(result);
+      });
+  }, []);
+  const [categories, setCategories] = useState([]);
+
+  const TopWrapper = styled.div`
+    position: absolute;
+    top: 91px;
```

Comment:
```suggestion
function SiteMap(props) {
  const [categories, setCategories] = useState([]);
  
  useEffect(() => {
    fetch('http://localhost:3000/data/navCategories.json')
      .then(response => response.json())
      .then(result => {
        const addedEmptyCategories = 2 * DIVIDE_CATEGORIES - result.length;
        if (
          result.length > DIVIDE_CATEGORIES &&
          result.length < 2 * DIVIDE_CATEGORIES
        ) {
          for (let i = 0; i < addedEmptyCategories; i++) {
            result.push({});
          }
        }
        setCategories(result);
      });
  }, []);
```
- for문을 고쳐보면 좋을거 같은데 혹시 어떤 로직인지 설명해주실 수 있나요?
- useState가 위로 useEffect가 아래로 오면 좋을거 같아요
- 컴포넌트 바깥으로 스타일드 컴포넌트 분리해주세요

Reply:
카테고리를 위쪽wrapper랑 아래쪽wrapper 
두 개로 나눴습니다.
그런데 두 개는 현재 6개로 되어있구요.
만약 아래쪽에 6개보다 부족한 3개나 2개가 들어오면
space-between에 따라서 카테고리 벌어집니다.
빈 카테고리를 추가시켜서 숫자를 맞추면 
위의 wrapper랑 아래 wrapper랑 세로로 각이 맞아집니다.